kore/src/main/java/org/kframework/utils/errorsystem/KException.java

```diff
@@ -103,7 +95,24 @@ public int hashCode() {
     }
 
     public enum ExceptionType {
-        ERROR, WARNING, HIDDENWARNING, NON_EXHAUSTIVE_MATCH, USELESS_RULE
+        ERROR,
+        NON_EXHAUSTIVE_MATCH,
+        UNDELETED_TEMP_DIR,
+        MISSING_HOOK_OCAML,
+        MISSING_SYNTAX_MODULE,
+        INVALID_EXIT_CODE,
+        DEPRECATED_BACKEND,
+        INVALID_CONFIG_VAR,
+        FUTURE_ERROR,
+        UNUSED_VAR,
+        PROOF_LINT,
+        FIRST_HIDDEN, // warnings below here are hidden by default
+        MISSING_HOOK_JAVA,
+        USELESS_RULE,
+        UNRESOLVED_FUNCTION_SYMBOL,
+        MALFORMED_MARKDOWN,
+        INVALIDATED_CACHE,
+        UNUSED_SYMBOL
```

Comment:
I think backend-specific things should not be in this list. So we should just have `MISSING_HOOK`, instead of `MISSING_HOOK_OCAML` or `MISSING_HOOK_JAVA`. And we should get rid of `DEPRECATED_BACKEND` and just use `FUTURE_ERROR`.

Reply:
I actually originally had these separate, but they had to be split because most java backend definitions report a lot of missing hook warnings just because a bunch of hooks were never implemented, and it makes the output extremely verbose whenever you call krun.

Reply:
So I had to make a separate category so I could make it disabled by default.

Reply:
Ah OK, makes sense.